### PR TITLE
Update unit tests so they don't litter console

### LINKED ISSUE #631: Swallow expected warnings in tests
At the moment, in the Travis CI logs, there are a lot of warnings being generated by unit tests.

These warnings are expected - e.g. a deprecated component.

It would be good if, where we expect warnings to be outputted from a unit test, we:

1. swallow the warning from stdout rather than letting it end up in travis logs
and 2. have the test check that a warning was logged

This may be doable by mocking console for a given test (ensuring to remove the mock when the test ends/fails/catches)

### PR DESCRIPTION
**Pull request by stevesims**
`console.warn` and `console.error` disabled in various tests

tweak to `top-nav` so that it no longer produces a warning

closes #631

**Checklist**:

* [ ] Documentation N/A
* [x] Tests
* [x] Ready to be merged

### CODE REVIEW

**Comment by penx** [comment]
Deploy preview for *govuk-react* processing.

Building with commit 79d9683d0f4d56717310fbfc46049b3648c0de26

https://app.netlify.com/sites/govuk-react/deploys/5c98cb6762785f0007d7d525

File: components/top-nav/src/index.js [hunk 1]
```diff
@@ -157,7 +157,7 @@ class TopNav extends Component {
                     <MenuButton open={this.state.navigationOpen} onClick={this.toggleNavigationOpen} />
                   </MenuButtonWrapper>
                   {/* Referenced in MenuButton */}
-                  <Input id="govuk-react-menu-button" type="checkbox" checked={this.state.navigationOpen} />
+                  <Input id="govuk-react-menu-button" type="checkbox" defaultChecked={this.state.navigationOpen} />
```
Comment (stevesims) [inline components/top-nav/src/index.js]: this change fixes an error shown in the console - which suggested either adding an `onClick` or changing this prop

i'm not entirely sure this is the right thing to do here though - as i don't understand native react form handling well enough - @penx thoughts?

Comment (penx) [reply components/top-nav/src/index.js]: hmm no I wouldn't change this to defaultChecked as pretty sure that would change the intended behaviour

**Comment by penx** [comment]
maybe revert the top nav change for now and open a new ticket to review the console message at some point

**Comment by penx** [comment]
PR good with me other than top nav change

**Comment by stevesims** [comment]
Topnav change reverted

**Pull request merged**